inference-engine/ie_bridges/java/tests/InferRequestTests.java

```diff
@@ -0,0 +1,85 @@
+import junit.framework.TestCase;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.intel.openvino.*;
+import org.intel.openvino.InferenceEngineProfileInfo.LayerStatus;
+
+public class InferRequestTests extends TestCase {
+    ConfigTests config = new ConfigTests();
+
+    IECore core;
+    CNNNetwork net;
+    ExecutableNetwork executableNetwork;
+    InferRequest inferRequest;
+    boolean completionCallback;
+
+    @Override
+    protected void setUp() {
+        try {
+            System.loadLibrary(IECore.NATIVE_LIBRARY_NAME);
+        } catch (UnsatisfiedLinkError e) {
+            System.err.println("Failed to load Inference Engine library\n" + e);
+            System.exit(1);
+        }
+        core = new IECore();
+        net = core.ReadNetwork(config.getXML());
+        executableNetwork = core.LoadNetwork(net, "CPU");
+        inferRequest = executableNetwork.CreateInferRequest();
+        completionCallback = false;
+    }
+
+    public void testGetPerformanceCounts(){   
+        inferRequest.Infer();
+        Map<String, InferenceEngineProfileInfo> expected = new HashMap<String, InferenceEngineProfileInfo>();
+        expected.put("19/Fused_Add_", new InferenceEngineProfileInfo(LayerStatus.EXECUTED, 0, 0, "jit_avx2_FP32", "Convolution", 0));
+        expected.put("21", new InferenceEngineProfileInfo(LayerStatus.NOT_RUN, 0, 0, "undef", "ReLU", 1));
+        expected.put("22", new InferenceEngineProfileInfo(LayerStatus.EXECUTED, 0, 0, "jit_avx_FP32", "Pooling", 2));
+        expected.put("23", new InferenceEngineProfileInfo(LayerStatus.EXECUTED, 0, 0, "jit_avx2_FP32", "Convolution", 3));
+        expected.put("24/Fused_Add_", new InferenceEngineProfileInfo(LayerStatus.EXECUTED, 0, 0, "jit_avx2_FP32", "Convolution", 4));
+        expected.put("26", new InferenceEngineProfileInfo(LayerStatus.NOT_RUN, 0, 0, "undef", "ReLU", 5));
+        expected.put("27", new InferenceEngineProfileInfo(LayerStatus.EXECUTED, 0, 0, "jit_avx_FP32", "Pooling", 6));
+        expected.put("29", new InferenceEngineProfileInfo(LayerStatus.EXECUTED, 0, 0, "jit_gemm_FP32", "FullyConnected", 7));
+        expected.put("fc_out", new InferenceEngineProfileInfo(LayerStatus.EXECUTED, 0, 0, "jit_avx2_FP32", "SoftMax", 8));
+        expected.put("out_fc_out", new InferenceEngineProfileInfo(LayerStatus.NOT_RUN, 0, 0, "unknown_FP32", "Output", 9));
```

Comment:
Do we really need a constructor for `InferenceEngineProfileInfo`? If it's only for this test - I'd like to recommend to rewrite the test. So we will have an array of reference layers types.

Reply:
I use it in GetPerformanceCounts method in c++ code.